### PR TITLE
Also render on error branch

### PR DESCRIPTION
**Pull request by sunng87**
@untitaker could you please check will this fit your requirement in #39 ?

### CODE REVIEW

File: src/middleware.rs [hunk 1]
```diff
@@ -159,6 +159,42 @@ impl AfterMiddleware for HandlebarsEngine {
             }
         }
     }
+
+    fn catch(&self, _: &mut Request, e: IronError) -> IronResult<Response> {
```
Comment (untitaker) [inline src/middleware.rs]: Can't you just do:

```
err.response = try!(self.after(request, err.response));
Err(err)
```

**Comment by sunng87** [comment]
Wow. That's better.

File: src/middleware.rs [hunk 2]
```diff
@@ -159,6 +159,12 @@ impl AfterMiddleware for HandlebarsEngine {
             }
         }
     }
+
+    fn catch(&self, req: &mut Request, e: IronError) -> IronResult<Response> {
```
Comment (untitaker) [inline src/middleware.rs]: `mut e: IronError`

**Comment by untitaker** [comment]
:+1: though, this fits my needs.

**Pull request merged**